internal/explorer.point/src/components/LinkPointToSol.jsx

```diff
@@ -0,0 +1,119 @@
+import { useState, useEffect } from 'react';
+import Swal from 'sweetalert2';
+import { useAppContext } from '../context/AppContext';
+
+const USER_REJECTION_CODE = 4001;
+const LS_KEY = 'point_to_sol_msg';
+const EMPTY_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+const styles = {
+    container: {
+        borderRadius: 4,
+        boxShadow: '0px 7px 15px 0 rgba(0, 0, 0, .15)',
+        padding: 16,
+        marginTop: 24,
+        color: 'rgb(124, 124, 124)',
+        display: 'grid',
+        gridTemplateColumns: '1fr 200px',
+        alignItems: 'center',
+        gridColumnGap: 42,
+    },
+    description: {
+        margin: 0,
+    },
+};
+
+const LinkPointToSol = () => {
+    const { walletIdentity } = useAppContext();
+    const [hasPointAddress, setHasPointAddress] = useState(true);
+    const [display, setDisplay] = useState(
+        () => localStorage.getItem(LS_KEY) !== 'hide',
+    );
+
+    useEffect(() => {
+        async function fetchData() {
+            try {
+                const { data } = await window.point.identity.identityToOwner({
+                    identity: walletIdentity,
+                });
+                if (!data.pointAddress || data.pointAddress === EMPTY_ADDRESS) {
+                    setHasPointAddress(false);
+                }
+            } catch (err) {
+                console.error(err);
+            }
+        }
+
+        if (
+            walletIdentity.endsWith('.sol') &&
+            window.point.point.link_point_to_sol
```

Comment:
**Minor:** the `fetchData` function does not depend on `window.point.point.link_point_to_sol`, but the `linkPointToSol` does. Also when dealing with function I think it is safer to check it like this `typeof window.point.point.link_point_to_sol === 'function'`.

Reply:
I didn't want `fetchData` to run if `window.point.point.link_point_to_sol` is not defined, because it would a waste of a network request (including a blockchain call).

Checking for `undefined` worked well on my tests. But no problem, I'll change it to `typeof`.

Reply:
Regarding `useCallback`, it is an unnecessary optimization in this case in my opinion. Defining a new function is a very cheap operation, and the only thing the function is doing is the data fetching. It is not used as a dependency anywhere else, and the component is in the homepage, which will not get many re-renders.

Generally speaking, I don't think it's a good idea to add `useCallback`, `useMemo` and other optimization techniques unless performance issues require it.